pkg/api/norman/customization/cluster/formatter.go

```diff
@@ -61,16 +45,45 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 	resource.Links["shell"] = shellLink
 	resource.AddAction(request, v32.ClusterActionGenerateKubeconfig)
 	resource.AddAction(request, v32.ClusterActionImportYaml)
+
+	// If user has permissions to update the cluster (regardless of RKE1 or not)
+	if canUpdateClusterWithValues(request, resource.Values) {
+		if convert.ToBool(resource.Values["enableClusterMonitoring"]) {
+			resource.AddAction(request, v32.ClusterActionDisableMonitoring)
+			resource.AddAction(request, v32.ClusterActionEditMonitoring)
+		} else {
+			resource.AddAction(request, v32.ClusterActionEnableMonitoring)
+		}
+	}
+
+	// If this is an RKE1 cluster only
 	if _, ok := resource.Values["rancherKubernetesEngineConfig"]; ok {
 		resource.AddAction(request, v32.ClusterActionExportYaml)
-		resource.AddAction(request, v32.ClusterActionRotateCertificates)
-		if rotateEncryptionKeyEnabled(f.clusterLister, resource.ID) {
-			resource.AddAction(request, v32.ClusterActionRotateEncryptionKey)
-		}
+
+		// If a user has the backupetcd role/privilege, add it
 		if _, ok := values.GetValue(resource.Values, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig"); ok {
-			resource.AddAction(request, v32.ClusterActionBackupEtcd)
-			resource.AddAction(request, v32.ClusterActionRestoreFromEtcdBackup)
+			if canBackupEtcd(request) {
+				resource.AddAction(request, v32.ClusterActionBackupEtcd)
+			}
+		}
+
+		// If user has permissions to update the cluster
+		if canUpdateClusterWithValues(request, resource.Values) {
+			if _, ok := values.GetValue(resource.Values, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig"); ok {
+				resource.AddAction(request, v32.ClusterActionRestoreFromEtcdBackup)
+			}
+			resource.AddAction(request, v32.ClusterActionRotateCertificates)
+			if rotateEncryptionKeyEnabled(f.clusterLister, resource.ID) {
+				resource.AddAction(request, v32.ClusterActionRotateEncryptionKey)
+			}
+
+			if val, ok := values.GetValue(resource.Values, "clusterTemplateRevisionId"); ok && val == nil {
+				if canCreateClusterTemplate(f.SubjectAccessReviewClient, request) {
+					resource.AddAction(request, v32.ClusterActionSaveAsTemplate)
```

Comment:
Before the change whether `v32.ClusterActionSaveAsTemplate` action is added or not was determined based on  `request.AccessControl.CanDo(v3.ClusterTemplateGroupVersionKind.Group, v3.ClusterTemplateResource.Name, "create", request, resource.Values, request.Schema)` check and now we're calling `canCreateClusterTemplate` which in turn is based on calling `CanCreateRKETemplate` which implementation looks quite different from `CanDo`.

Could you please confirm these 2 approaches are equivalent or we're fixing some discrepancy here?

Reply:
That's interesting. We use rbac in `formatter.go` but our international authenticator on the HTTP request header in `canCreateClusterTemplate.` Looking into it...

Reply:
Ok, so looks like the first implementation in `formatter.go` uses a framework in `Steve` and the second one uses built-in Kubernetes. Going to do some debugging to make sure the roles and role bindings that are checked are the same.

Reply:
Synced with @thedadams real quick about this and I think the best course of action is to just revert the call to `canCreateClusterTemplate` in `formatter.go` back to the RBAC call to avoid a regression. Let me know if this looks good